src/app.js

```diff
@@ -0,0 +1,17 @@
+const express = require('express');
+require('dotenv').config();
+
+const connectToMongo = require('../db/connection');
+
+const app = express();
+const port = process.env.PORT;
+
+app.use(express.urlencoded({ extended: false }));
+app.use(express.json());
+
+app.listen(port, () => {
```

Comment:
Here, before listening, you can check the environment, and only listen if the environment is not "test". Something like: 
```
if (process.env.NODE_ENV !== 'test') {
  app.listen(port, () => {
    console.log(`Server listening on port ${port}`);
    connectToMongo();
  });
}
````